iot_verifier/src/reward_share.rs

```diff
@@ -205,28 +351,65 @@ impl GatewayShares {
                         witness_rewards_per_share,
                         reward_shares.witness_shares,
                     ),
+                    dc_transfer_amount: compute_rewards(
+                        dc_transfer_rewards_per_share,
+                        reward_shares.dc_shares,
+                    ),
                     start_period: reward_period.start.encode_timestamp(),
                     end_period: reward_period.end.encode_timestamp(),
                 },
             )
             .filter(|reward_share| {
-                reward_share.beacon_amount > 0 || reward_share.witness_amount > 0
+                reward_share.beacon_amount > 0
+                    || reward_share.witness_amount > 0
+                    || reward_share.dc_transfer_amount > 0
             })
     }
 }
 
+/// returns the equiv iot bones value for a specified dc amount
+pub fn dc_to_iot_bones(dc_amount: Decimal, iot_price: Decimal) -> Decimal {
+    // iot prices are supplied in 10^6
+    // need to do same for dc usd value
+    let dc_in_usd = dc_amount * (*DC_USD_PRICE);
+    (dc_in_usd * dec!(1_000_000) / iot_price)
+        .round_dp_with_strategy(DEFAULT_PREC, RoundingStrategy::ToPositiveInfinity)
+}
+
+/// returns the equiv dc value for a specified iot bones amount
+pub fn iot_bones_to_dc(iot_amount: Decimal, iot_price: Decimal) -> Decimal {
+    // iot is in bones ( 10^6)
+    // account for that when determining dc amount
+    let iot_value = iot_amount * iot_price;
+    (iot_value / (*DC_USD_PRICE) / dec!(1_000_000))
+        .round_dp_with_strategy(0, RoundingStrategy::ToNegativeInfinity)
+}
+
+pub fn normalize_dc_transfer_rewards(
+    total_dc_transfer_rewards_used: Decimal,
+    total_dc_transfer_rewards: Decimal,
+) -> (Decimal, Decimal) {
+    match total_dc_transfer_rewards_used < total_dc_transfer_rewards {
```

Comment:
any specific reason this _shouldn't_ be `<=`? we won't have any dangling fractional amounts unaccounted for by leaving that scenario unaccounted for?

Reply:
addressed here: 6b9ee2328eae198a9a1e7426b775a8f7ca85bf92